Assignment-11/src/classes/ContactTriggerHandlerTest.cls

```diff
@@ -0,0 +1,228 @@
+/** 
+ * Purpose          :   Test Class for contactTriggerHandler
+ * 
+ * Created by       :   Sajal Mittal
+ * 
+ * Created Date     :   13-12-2024
+ * 
+ * Rivision Logs    :   V_1.0 - Created
+ * 
+**/
+@isTest(seeAllData = FALSE)
+private class ContactTriggerHandlerTest {
+
+    // Test Setup method for creating pre-required fake data
+    @TestSetup
+    static void makeData(){
+        List<Account> accList = new list<Account>();
+        accList.add(new Account(Name = 'Test Account 1', Phone = '0000000000', Domain__c='test1.com'));
+        accList.add(new Account(Name = 'Test Account 2', Phone = '0000000000', Domain__c='test2.com'));
+        accList.add(new Account(Name = 'Test Account 3', Phone = '0000000000'));
+        insert accList;
+        
+        List<Contact> contactsToInsert = createContacts();
+        Database.insert(contactsToInsert, false);
+    }
+    
+    /**
+     * Test Method for testing Insertion of Contact record based on Email 
+     * if it is associated with an Account and have correct Domain as on Account.
+     **/
+    @isTest
+    static void myUnitTestForInsertion() {
+        List<Contact> contactsToInsert = createContacts();
+        
+        Test.startTest();
+        insert contactsToInsert[1];
```

Comment:
One tab (Four Space) gap should be present in between Test.startTest() and Test.stopTest().